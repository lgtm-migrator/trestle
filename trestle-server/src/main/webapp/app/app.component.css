html {
    height: 100%;
}
body {
    margin: 0;
    text-align: center;
    min-height: 100%;
}