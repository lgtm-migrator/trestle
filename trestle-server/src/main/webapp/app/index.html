<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <title>Trestle</title>
    <base href="/workspace/">
</head>
<body>
<app-root>
    loading...
</app-root>
</body>
</html>
