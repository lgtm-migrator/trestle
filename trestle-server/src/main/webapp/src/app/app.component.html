<mat-toolbar color="primary" class="top-layout z-depth-4">
  <h4 (click)="sidenav.toggle()">TRESTLE</h4>
  <span class="toolbar-spacer"></span>
  <!--User login specific functions-->
  <div class="action-buttons" *ngIf="user | async as user; else loginDiv">
    <img class="img-circle" [src]="getGravatarURL()" alt="User profile picture">
    <button id="logout" mat-button class="toolbar-button" (click)="logout()" aria-label="Logout">
      <fa-icon [icon]="logoutIcon" [size]="iconSize"></fa-icon>
    </button>
  </div>
  <ng-template #loginDiv>
    <button id="login" mat-button class="toolbar-button" routerLink="/login" aria-label="Login">
      <fa-icon [icon]="loginIcon" [size]="iconSize"></fa-icon>
    </button>
  </ng-template>
</mat-toolbar>
<mat-sidenav-container>
  <mat-sidenav #sidenav class="z-depth-3" [mode]="mobileQuery.matches ? 'over' : 'side'"
               [fixedInViewport]="mobileQuery.matches" fixedTopGap="56" [opened]="!mobileQuery.matches">
    <mat-nav-list *ngIf="user | async as user">
      <div id="user-actions">
        <h5 matSubheader>Explore</h5>
        <mat-divider></mat-divider>
        <a id="viewer" mat-list-item [routerLink]="['explore/viewer']" aria-label="dataset viewer link">
          <mat-icon>map</mat-icon>
          VIEWER</a>
        <a id="visualize" mat-list-item [routerLink]="['explore/visualize']" aria-label="visualizer link">
          <mat-icon>view_carousel</mat-icon>
          VISUALIZER</a>
        <a id="query" mat-list-item [routerLink]="['explore/query']" aria-label="query link">
          <mat-icon>query_builder</mat-icon>
          QUERY</a>
        <a id="compare" mat-list-item [routerLink]="['explore/compare']" aria-label="compare links">
          <mat-icon>compare</mat-icon>
          COMPARE</a>
        <a id="aggregate" mat-list-item [routerLink]="['explore/aggregate']" aria-label="aggregate links">
          <mat-icon>call_merge</mat-icon>
          AGGREGATE</a>
      </div>
      <div id="admin-actions" *ngIf="userHasRequiredPermissions(user,[Privileges.ADMIN])">
        <mat-divider></mat-divider>
        <h5 matSubheader>Admin</h5>
        <mat-divider></mat-divider>
        <a id="dashboard" mat-list-item [routerLink]="['admin/dashboard']" aria-label="dashboard link">
          <mat-icon>dashboard</mat-icon>
          DASHBOARD</a>
        <a id="metrics" mat-list-item [routerLink]="['admin/metrics']" aria-label="metrics link">
          <mat-icon>show_chart</mat-icon>
          METRICS</a>
        <a id="users" mat-list-item [routerLink]="['admin/users']" aria-label="users link">
          <mat-icon>supervisor_account</mat-icon>
          USERS</a>
      </div>
      <div id="dba-actions" *ngIf="userHasRequiredPermissions(user,[Privileges.DBA])">
        <a id="index" mat-list-item [routerLink]="['admin/index']" aria-label="index link">
          <mat-icon>supervisor_account</mat-icon>
          INDICIES</a>
      </div>
      <mat-divider></mat-divider>
    </mat-nav-list>
  </mat-sidenav>
  <div class="app-content">
    <router-outlet></router-outlet>
  </div>
</mat-sidenav-container>
