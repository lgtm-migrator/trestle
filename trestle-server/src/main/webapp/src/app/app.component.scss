html {
  height: 100%;
}
.app-content {
  min-height: 100vh;
}
