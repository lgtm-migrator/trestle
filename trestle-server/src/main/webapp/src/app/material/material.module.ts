import {NgModule} from '@angular/core';
import {MatAutocompleteModule} from '@angular/material/autocomplete';
import {MatButtonToggleModule} from '@angular/material/button-toggle';
import {MatExpansionModule} from '@angular/material/expansion';
import {CdkTableModule} from '@angular/cdk/table';
import {MatButtonModule} from '@angular/material/button';
import {MatCardModule} from '@angular/material/card';
import {MatCheckboxModule} from '@angular/material/checkbox';
import {MatChipsModule} from '@angular/material/chips';
import {MatDialogModule} from '@angular/material/dialog';
import {MatFormFieldModule} from '@angular/material/form-field';
import {MatGridListModule} from '@angular/material/grid-list';
import {MatIconModule} from '@angular/material/icon';
import {MatInputModule} from '@angular/material/input';
import {MatListModule} from '@angular/material/list';
import {MatMenuModule} from '@angular/material/menu';
import {MatOptionModule} from '@angular/material/core';
import {MatProgressBarModule} from '@angular/material/progress-bar';
import {MatProgressSpinnerModule} from '@angular/material/progress-spinner';
import {MatRadioModule} from '@angular/material/radio';
import {MatSelectModule} from '@angular/material/select';
import {MatSidenavModule} from '@angular/material/sidenav';
import {MatSliderModule} from '@angular/material/slider';
import {MatSlideToggleModule} from '@angular/material/slide-toggle';
import {MatTableModule} from '@angular/material/table';
import {MatTabsModule} from '@angular/material/tabs';
import {MatToolbarModule} from '@angular/material/toolbar';
import {MatTooltipModule} from '@angular/material/tooltip';
import {MatPaginatorModule} from '@angular/material/paginator';

@NgModule({
  imports: [
    CdkTableModule,
    MatAutocompleteModule,
    MatButtonModule,
    MatButtonToggleModule,
    MatCardModule,
    MatCheckboxModule,
    MatChipsModule,
    MatDialogModule,
    MatExpansionModule,
    MatFormFieldModule,
    MatGridListModule,
    MatIconModule,
    MatInputModule,
    MatListModule,
    MatMenuModule,
    MatOptionModule,
    MatPaginatorModule,
    MatProgressBarModule,
    MatProgressSpinnerModule,
    MatRadioModule,
    MatSelectModule,
    MatSidenavModule,
    MatSliderModule,
    MatSlideToggleModule,
    MatTableModule,
    MatTabsModule,
    MatToolbarModule,
    MatTooltipModule],
  exports: [
    CdkTableModule,
    MatAutocompleteModule,
    MatButtonModule,
    MatButtonToggleModule,
    MatCardModule,
    MatCheckboxModule,
    MatChipsModule,
    MatDialogModule,
    MatExpansionModule,
    MatFormFieldModule,
    MatGridListModule,
    MatIconModule,
    MatInputModule,
    MatListModule,
    MatMenuModule,
    MatOptionModule,
    MatPaginatorModule,
    MatProgressBarModule,
    MatProgressSpinnerModule,
    MatRadioModule,
    MatSelectModule,
    MatSidenavModule,
    MatSliderModule,
    MatSlideToggleModule,
    MatTableModule,
    MatTabsModule,
    MatToolbarModule,
    MatTooltipModule]
})
export class MaterialModule {
}
