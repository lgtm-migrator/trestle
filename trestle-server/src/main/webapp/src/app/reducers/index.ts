import {ActionReducerMap, createReducer, createSelector, MetaReducer, on} from '@ngrx/store';
import {environment} from '../../environments/environment';
import {TrestleUser} from '../user/trestle-user';
import {login, loginFailure, loginSuccess, logout} from '../actions/auth.actions';

export interface UserState {
  user: TrestleUser;
  userError: Error;
  userToken: string;
}

const initialUserState: UserState = {
  user: null,
  userError: null,
  userToken: ''
};

export interface State {
  user: UserState;
}

const _authReducer = createReducer(initialUserState, on(login, state => {
    console.log('Trying to login');
    return state;
  }),
  on(loginSuccess, (state, {user, token}) => ({
    userError: null,
    user,
    userToken: token
  })),
  on(loginFailure, (state, {error}) => ({
    user: null,
    userError: error,
    userToken: ''
  })),
  on(logout, (state) => ({
    ...state,
    user: null,
    token: ''
  })));


export function authReducer(state: UserState = initialUserState, action): UserState {
  return _authReducer(state, action);
}

export const reducers: ActionReducerMap<State> = {
  user: authReducer
};


export const metaReducers: MetaReducer<State>[] = !environment.production ? [] : [];


export const selectUserState = (state: State) => state.user;
export const selectUser = (state: UserState) => state.user;


export const selectUserFromUser = createSelector(
  selectUserState,
  selectUser);

export const selectErrorFromUser = createSelector(
  selectUserState,
  (state) => state.userError
);

export const selectTokenFromUser = createSelector(
  selectUserState,
  (state) => state.userToken
);
