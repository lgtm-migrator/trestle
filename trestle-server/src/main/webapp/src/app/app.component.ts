/**
 * Created by nrobison on 1/19/17.
 */
import {ChangeDetectorRef, Component, OnDestroy, OnInit, ViewEncapsulation} from '@angular/core';
import {TrestleUser} from './user/trestle-user';
import {AuthService, Privileges} from './user/authentication.service';
import {Router} from '@angular/router';
import {MD5} from 'crypto-js';
import {faSignInAlt, faSignOutAlt} from '@fortawesome/free-solid-svg-icons';
import {SizeProp} from '@fortawesome/fontawesome-svg-core';
import {MediaMatcher} from '@angular/cdk/layout';
import {select, Store} from '@ngrx/store';
import {State} from './reducers';
import {Observable} from 'rxjs';
import {tap} from 'rxjs/operators';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
  encapsulation: ViewEncapsulation.None
})
export class AppComponent implements OnInit, OnDestroy {
  public readonly iconSize: SizeProp = 'lg';
  public readonly loginIcon = faSignInAlt;
  public readonly logoutIcon = faSignOutAlt;
  public gravatarURL: string;
  // We need this in order to access the Privileges enum from the template
  public Privileges = Privileges;
  public user: Observable<TrestleUser>;

  public mobileQuery: MediaQueryList;
  private readonly mobileQueryListener: () => void;

  constructor(private authService: AuthService,
              private router: Router,
              private store: Store<State>,
              changeDetectorRef: ChangeDetectorRef,
              media: MediaMatcher
  ) {
    this.mobileQuery = media.matchMedia('(max-width: 800px)');
    this.mobileQueryListener = () => changeDetectorRef.detectChanges();
    this.mobileQuery.addEventListener('change', this.mobileQueryListener);
  }


  public ngOnInit(): void {
    // Get the current user, if it exists
    this.user = this.store.pipe(select("user"), select("user"), tap(user => console.log("User: ", user)));
  }

  public ngOnDestroy(): void {
    this.mobileQuery.removeEventListener('change', this.mobileQueryListener);
    // this.loginSubscription.unsubscribe();
  }

  // /**
  //  * Attempt to login the user
  //  */
  // public login(): void {
  //   if (this.user == null) {
  //     this.router.navigate(['/login']);
  //   }
  // }

  /**
   * Does the user have the required permissions?
   * @param {TrestleUser} user to verify permisisons on
   * @param {Privileges[]} requiredPrivs
   * @returns {boolean}
   */
  public userHasRequiredPermissions(user: TrestleUser, requiredPrivs: Privileges[]): boolean {
    return user.hasRequiredPrivileges(requiredPrivs);
  }

  /**
   * Get the Gravitar URL of the user
   * @returns {string}
   */
  public getGravatarURL(): string {
    return "";
    // if (this.gravatarURL == null) {
    //   const user = this.authService.getUser();
    //   if (user !== null) {
    //     const hash = MD5(user.email.trim().toLowerCase()).toString();
    //     this.gravatarURL = 'https://www.gravatar.com/avatar/' + hash + '?d=identicon' + '&s=36';
    //     return this.gravatarURL;
    //   }
    // }
    // return this.gravatarURL;
  }

  /**
   * Logout the currently logged in user
   */
  public logout(): void {
    this.authService.logout();
    this.user = null;
  }
}
