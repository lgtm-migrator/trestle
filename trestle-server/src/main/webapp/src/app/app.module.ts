import {BrowserModule} from '@angular/platform-browser';
import {NgModule} from '@angular/core';

import {AppRoutingModule} from './app-routing.module';
import {AppComponent} from './app.component';
import {HttpClientModule} from '@angular/common/http';
import {BrowserAnimationsModule} from '@angular/platform-browser/animations';
import {UserModule} from './user/user.module';
import {MaterialModule} from './material/material.module';
import {NavigationModule} from './navigation/navigation.module';
import {FontAwesomeModule} from '@fortawesome/angular-fontawesome';
import {JWT_OPTIONS, JwtModule} from '@auth0/angular-jwt';
import {environment} from '../environments/environment';
import {AuthService} from './user/authentication.service';
import { StoreModule } from '@ngrx/store';
import { reducers, metaReducers } from './reducers';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';
import { EffectsModule } from '@ngrx/effects';
import { AuthEffects } from './effects/auth.effects';

export function jwtOptionsFactory(service: AuthService) {
  // noinspection JSUnusedGlobalSymbols
  return {
    tokenGetter: () => {
      return service.getEncodedToken();
    },
    whitelistedDomains: [environment.domain],
    blacklistedRoutes: ['http://localhost:8080/auth/login'],
    throwNoTokenError: true,
    authScheme: ''
  };
}


@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    HttpClientModule,
    AppRoutingModule,
    BrowserModule,
    BrowserAnimationsModule,
    UserModule,
    MaterialModule,
    NavigationModule,
    FontAwesomeModule,
    JwtModule.forRoot({
      jwtOptionsProvider: {
        provide: JWT_OPTIONS,
        useFactory: jwtOptionsFactory,
        deps: [AuthService]
      }
    }),
    StoreModule.forRoot(reducers, {
      metaReducers, 
      runtimeChecks: {
        strictStateImmutability: true,
        strictActionImmutability: true,
      }
    }),
    !environment.production ? StoreDevtoolsModule.instrument() : [],
    EffectsModule.forFeature([AuthEffects]),
    EffectsModule.forRoot([])
  ],
  bootstrap: [AppComponent]
})
export class AppModule {
}
