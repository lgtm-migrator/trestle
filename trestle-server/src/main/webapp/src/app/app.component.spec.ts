import {async, TestBed} from '@angular/core/testing';
import {RouterTestingModule} from '@angular/router/testing';
import {AppComponent} from './app.component';
import {MaterialModule} from "./material/material.module";
import {AuthService} from "./user/authentication.service";
import {HttpClientTestingModule} from "@angular/common/http/testing";
import {NoopAnimationsModule} from "@angular/platform-browser/animations";
import {FontAwesomeModule} from '@fortawesome/angular-fontawesome';

describe('AppComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        RouterTestingModule,
        HttpClientTestingModule,
        NoopAnimationsModule,
        MaterialModule,
        FontAwesomeModule
      ],
      providers: [AuthService],
      declarations: [
        AppComponent
      ],
    }).compileComponents();
  }));

  it('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    expect(fixture).toMatchSnapshot();
  });
});
