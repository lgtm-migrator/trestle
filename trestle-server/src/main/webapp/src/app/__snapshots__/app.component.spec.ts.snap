// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should create the app 1`] = `
<app-root
  Privileges={[Function Object]}
  authService={[Function AuthService]}
  mobileQuery={[Function Object]}
  mobileQueryListener={[Function Function]}
  router={[Function Router]}
  store={[Function MockStore]}
>
  <top-nav>
    <mat-toolbar
      class="top-layout z-depth-4 mat-toolbar"
      color="primary"
    >
      <h4>
        TRESTLE
      </h4>
      <span
        class="toolbar-spacer"
      />
      <user-icon>
        
        
      </user-icon>
    </mat-toolbar>
  </top-nav><mat-sidenav-container
    class="mat-drawer-container mat-sidenav-container"
  >
    
    
  </mat-sidenav-container>
</app-root>
`;
